### DISCUSSION TITLE
Generation of Tags, Classifier and SWID Tag ID along with project without UI for DevOps Automation

### DISCUSSION THREAD

**Discussion by aejazuddinam** [DependencyTrack/dependency-track#1156] [Q&A]
Hi Steve/Team,

Do we have option from Jenkins pipeline parameters to add Tags, Classifier and SWID Tag ID or any other identity when creating the project, or Post Via REST APIs (majorly needed tags & swid tag id), because i see only creation of project, its version and uploading of Bom from jenkins parameters and via /v1/bom api (post). I need this as requirement is for automation from pipelines without involving the dashboard UI.

**Comment by stevespringett** [comment]
The DT API absolutely supports these capabilities, post project creation, as that's what the frontend leverages. I don't know if the Jenkins plugin supports this or not. You may want to reach out the maintainer. Jenkins and all plugins use a dedicated JIRA instance. I'd recommend creating an enhancement request if the plugin doesn't already support it. 

https://plugins.jenkins.io/dependency-track/

**Reply by aejazuddinam** [reply]
Thanks Steve, I've gone through the REST API docs https://docs.dependencytrack.org/integrations/rest-api/
and followed all the APIs given in Swagger , http://{myhost}:{myport}/api/swagger.json

But Post methods are given with endpoint /v1/bom for creating a project ,version, uploading bom and some other things, but there are no post methods for creating new tags , assign a classifier and creating SWID Tag ID . Kindly help me if im missing anything. 

And for Jenkins Plugin i will raise a request with the Plugin maintainer for enhancement.

**Reply by stevespringett** [reply]
When creating a new project using PUT `/v1/project` or updating an existing project using  POST `/v1/project` you can specify `classifier`, `cpe`, `purl`, and `swidTagId`. You can also specify `tags`. If you want specifics, the best way is to open up developer tools in your web browser, go to the network tab, and create or update a project with the data you want. You'll see the request the UI makes.

The resources at `/v1/bom` cannot be used to modify projects.
